app/javascript/src/BackendApis/components/AddBackendForm.jsx

```diff
@@ -0,0 +1,73 @@
+// @flow
+
+import * as React from 'react'
+import { useState } from 'react'
+
+import {
+  Form,
+  ActionGroup,
+  Button,
+  PageSection,
+  PageSectionVariants
+} from '@patternfly/react-core'
+import { BackendSelect, PathInput } from 'BackendApis'
+import { CSRFToken } from 'utilities/utils'
+import { createReactWrapper } from 'utilities/createReactWrapper'
+
+import type { Backend } from 'Types'
+
+import './AddBackendForm.scss'
+
+type Props = {
+  backends: Backend[],
+  url: string,
+  newBackendPath: string
+}
+
+const AddBackendForm = ({ backends, url, newBackendPath }: Props): React.Node => {
+  const [backend, setBackend] = useState<Backend | null>(null)
+  const [path, setPath] = useState('')
+  const [loading, setLoading] = useState(false)
+
+  const isFormComplete = backend !== null && path !== ''
```

Comment:
A component re-renders after every change in the state, therefore `isFormComplete` will be re-calculated every time. Since its value depends directly on the inputs, it doesn't belong in the State.